mockgcp/mocklogging/logmetric.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mocklogging
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/mockgcp/common/projects"
+	pb "github.com/GoogleCloudPlatform/k8s-config-connector/mockgcp/generated/mockgcp/logging/v2"
+	"github.com/golang/protobuf/ptypes/empty"
+)
+
+type metricsService struct {
+	*MockService
+	pb.UnimplementedMetricsServiceV2Server
+}
+
+func (s *metricsService) GetLogMetric(ctx context.Context, req *pb.GetLogMetricRequest) (*pb.LogMetric, error) {
+	name, err := s.parseLogMetricName(req.MetricName)
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := name.String()
+
+	obj := &pb.LogMetric{}
+	if err := s.storage.Get(ctx, fqn, obj); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, status.Errorf(codes.NotFound, "metric %q not found", name)
+		} else {
+			return nil, status.Errorf(codes.Internal, "error reading metric: %v", err)
+		}
+	}
+
+	return obj, nil
+}
+
+func (s *metricsService) CreateLogMetric(ctx context.Context, req *pb.CreateLogMetricRequest) (*pb.LogMetric, error) {
+	reqName := req.Parent + "/metrics/" + req.GetMetric().GetName()
+	name, err := s.parseLogMetricName(reqName)
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := name.String()
+
+	now := time.Now()
+
+	obj := proto.Clone(req.GetMetric()).(*pb.LogMetric)
+	obj.Name = name.MetricName
+	obj.CreateTime = timestamppb.New(now)
+	obj.UpdateTime = timestamppb.New(now)
+
+	// Not clear why, but we return the values before populating the MetricDescriptor Name/Type
```

Comment:
/we return the values/we construct the return values/

Reply:
That is weird, especially for a mock. Timestamps can sometimes be sensitive for security.

Reply:
I'll rephrase, but what I wrote is correct.  The value we return should not have these default values set.  When we get it immediately afterwards, the values are there.  (And there's no LRO in this API either, which is also odd)

Reply:
*Pedant alert*
I agree your return value does not have the values. However your not returning before you populate the values. Your constructing the return value before you populate.

Reply:
Saying "we return ... before" makes me think its talking about a return statement before some other action. 
As that isn't what its talking about I think its confusing.